src/averageAges.js

```diff
@@ -1,61 +1,82 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = people.filter(person => person.sex === 'm');
+
+  men = century
+    ? men.filter(person => Math.ceil(person.died / 100) === century)
+    : men;
+
+  const sumOfAgeMan = calculateSumOfAvarageAge(men);
+  const averageAgeOfMen = calculateAverage(sumOfAgeMan, men.length);
+
+  return averageAgeOfMen;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = withChildren
+    ? people.filter(person => person.mother !== null).map(el => el.mother)
+    : people.filter(person => person.sex === 'f');
+
+  women = withChildren
+    ? people.filter(el => women.includes(el.name))
+    : women;
```

Comment:
If you chain a few methods in a row, it is better to start a new method from a new line